### ISSUE TITLE
Kernel SVM training with data in a fixed sequence

### ISSUE THREAD

**Issue by Gourmentic** [nfmcclure/tensorflow_cookbook#158]
Hello, I find a very interesting thing: after changing the code of kernel SVM as below, a better training progress and a better performance is shown.

#rand_index = np.random.choice(len(x_vals), size=batch_size)
#rand_x = x_vals[rand_index]
#rand_y = np.transpose([y_vals[rand_index]])
rand_x = x_vals
rand_y = y_vals.reshape(-1, 1)

![image](https://user-images.githubusercontent.com/23225921/50078219-5f721880-0221-11e9-889b-c5b60345c2ed.png)
![image](https://user-images.githubusercontent.com/23225921/50078259-79136000-0221-11e9-851f-3f84d4265523.png)
![image](https://user-images.githubusercontent.com/23225921/50078358-b8da4780-0221-11e9-82e3-549486b43c26.png)
![image](https://user-images.githubusercontent.com/23225921/50078370-c263af80-0221-11e9-8c61-9432cfcd9297.png)

I think the reason is that every scalar bi in Vector Variable b should follow the ith data pair(feature,label) according to the dual formula. But random choice makes the weight bi be trained by different data pair ,not only the ith and maybe give a wrong direction to update bi. 
![image](https://user-images.githubusercontent.com/23225921/50078666-8715b080-0222-11e9-8e08-a6e7e55142a5.png)

And tatnguyennguyen give the reason why the current code get a good result. 
https://github.com/nfmcclure/tensorflow_cookbook/issues/104

So the current code is only ok for the whole-batch training. Maybe embedding layer would be used for a mini batch training. :)

**Comment by anbo1024** [comment]
@Gourmentic 
I also found this problem. Have you solved it?

**Comment by Gourmentic** [comment]
SGD can not be used as optimizer of Kernel SVM directly, the mentioned code in the book should be changed.
